.github/workflows/ci.yml

```diff
@@ -45,6 +45,52 @@ jobs:
         run: |
           make -C build/ help | grep demo | tr -d '. ' | xargs make -C build/
           make -C demos/jobs/jobs_demo_mosquitto
+  build-check-demos-cpp:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Rename source files
+        run: |
+          for demo in mqtt http shadow defender
+          do
+              for source in demos/$demo/*/*.c
+              do
+                  mv -- "$source" "${source%.c}.cpp"
+              done
+          done
+      - name: Configure CMake build
+        run: |
+          sudo apt-get install -y libmosquitto-dev
+          curl https://cmake.org/files/v3.2/cmake-3.2.0-Linux-x86_64.tar.gz -o cmake.tar.gz
+          tar -xf cmake.tar.gz
+          mkdir build && cd build
+          ../cmake-3.2.0-Linux-x86_64/bin/cmake .. \
+          -G "Unix Makefiles" \
+          -DBUILD_DEMOS=1 \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_C_FLAGS='-Wall -Wextra' \
+          -DAWS_IOT_ENDPOINT="aws-iot-endpoint" \
+          -DBROKER_ENDPOINT="broker-endpoint" \
+          -DCLIENT_CERT_PATH="cert/path" \
+          -DROOT_CA_CERT_PATH="cert/path" \
+          -DCLIENT_PRIVATE_KEY_PATH="key/path" \
+          -DCLIENT_IDENTIFIER="ci-identifier" \
+          -DTHING_NAME="thing-name" \
+          -DS3_PRESIGNED_GET_URL="get-url" \
+          -DS3_PRESIGNED_PUT_URL="put-url" \
+          -DCLAIM_CERT_PATH="cert/path" \
+          -DCLAIM_PRIVATE_KEY_PATH="key/path" \
+          -DPROVISIONING_TEMPLATE_NAME="template-name" \
+          -DPROVISIONING_CSR_PATH="csr/path" \
+          -DPROVISIONING_PRIVATE_KEY_PATH="key/path" \
+          -DPROVISIONING_CERT_PATH="cert/path"
+      - name: Build Demos
+        run: |
+          make -C build/ help | grep demo | tr -d '. ' | xargs make -C build/
+          make -C demos/jobs/jobs_demo_mosquitto
```

Comment:
Seems like this is not required as Jobs is not tested for C++ compliance?

Reply:
Thanks, copypasta from the other job